### DISCUSSION TITLE
Is it possible to disable putting speaker names in Whisper output?

### DISCUSSION THREAD

**Discussion by Arderos** [openai/whisper#854] [Q&A] [answered]
Hi!
I'm using Whisper to generate subtitles to our youtube show and it works really well. There is one thing that is not particularly convenient. Sometimes Whisper is trying to identify the speakers and put their names into the resulting output. Is there any way to disable that?

**Comment by jongwook** [comment, marked as answer]
The models have been fine-tuned to prevent this, but apparently it didn't prevent all cases. Unfortunately there is no perfect way to achieve this, but if you see certain patterns for the speaker names, such as `[NAME]` or `NAME:`, you could use a regex filter to remove those.

The decoding implementation by default suppresses letters like `[`, `]`, `:` in the transcript:

https://github.com/openai/whisper/blob/0f39c89d9212e4d0c64b915cf7ba3c1f0b59fecc/whisper/tokenizer.py#L237-L255

But it might be helpful to enable those in order to filter them out post-hoc.

**Reply by Arderos** [reply]
Thanks for the suggestion! I should have pointed out how the output looks like exactly. It looks like so: "Some spoken text. Another spoken text. Name. Yet another spoken text."
